app/controllers/sks_users_controller.ts

```diff
@@ -0,0 +1,121 @@
+import { HttpContext } from '@adonisjs/core/http'
+import SksUser from '#models/sks_user'
+import { DateTime } from 'luxon'
+
+//this value determines the period over which the trend will be counted (multiply the value by 5 minutes)
+const trendDelta = 3
+
+enum Trend {
+  INCREASING = 'INCREASING',
+  DECREASING = 'DECREASING',
+  STABLE = 'STABLE',
+}
+
+export default class SksUsersController {
+  /**
+   * Display the latest SKS users record based on time
+   */
+  async latest({ response }: HttpContext) {
+    try {
+      const currentTime = DateTime.now().setZone('Europe/Warsaw').toSQL()
+      if (currentTime === null) {
+        return response.status(500).json({ message: 'Failed to convert time to SQL format' })
+      }
+
+      let latestData = await SksUser.query()
+        .where('externalTimestamp', '<', currentTime)
+        .orderBy('externalTimestamp', 'desc')
+        .first()
+      let isResultRecent = true
+
+      // If the first record has activeUsers set to 0, get the second record instead
+      if (latestData !== null && latestData.activeUsers === 0) {
+        latestData = await SksUser.query()
+          .where('externalTimestamp', '<', currentTime)
+          .orderBy('externalTimestamp', 'desc')
+          .offset(1)
+          .first()
+        isResultRecent = false
+      }
+
+      if (latestData === null) {
+        return response
+          .status(404)
+          .json({ message: 'Could not find the matching data in database' })
+      }
+
+      const referenceTime = latestData.externalTimestamp?.toSQL() || ''
+      if (!referenceTime) {
+        return response
+          .status(500)
+          .json({ message: 'Failed to convert external timestamp to SQL format' })
+      }
+
+      const trend = await this.calculateTrend(latestData, referenceTime, trendDelta)
+
+      return response
+        .status(200)
+        .json({ ...latestData.toJSON(), trend: trend, isResultRecent: isResultRecent })
```

Comment:
you can do it shorter with  `{ ...latestData.toJSON(), trend, isResultRecent }` syntax

Reply:
that's cool